### PR TITLE
[stdlib] Updating LibraTimestamp and other changes.

### PR DESCRIPTION
**Pull request by wrwg**
This changes the definition of `is_genesis` to not depend on acquiring the CurrentTime resource. There are also multiple smaller
changes here and there (using CoreAddresses also in specs and others). Also changes the doc generator to distinguish
resources from structs, which leads to larger diff in generated docs.

Submitting this now to not accumulate to many merge conflicts.

## Motivation

Specs

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Existing tests

## Related PRs

NA

### CODE REVIEW

File: language/stdlib/modules/LibraTimestamp.move [hunk 1]
```diff
@@ -19,7 +33,19 @@ module LibraTimestamp {
         move_to(association, timer);
     }
 
-    // Update the wall clock time by consensus. Requires VM privilege and will be invoked during block prologue.
+    /// Marks that time has started and genesis has finished. This can only be called from genesis.
+    public fun set_time_has_started(association: &signer) acquires CurrentTimeMicroseconds {
+        assert(Signer::address_of(association) == CoreAddresses::ASSOCIATION_ROOT_ADDRESS(), 1);
+
+        // Current time must have been initialized.
+        assert(exists<CurrentTimeMicroseconds>(
```
Comment (sblackshear) [inline language/stdlib/modules/LibraTimestamp.move:41]: Nit: this formatting is somewhat hard to read

Comment (wrwg) [reply language/stdlib/modules/LibraTimestamp.move:41]: Done

**Pull request closed without merging**